pkg/url/url.go

```diff
@@ -1086,3 +1125,38 @@ func Push(client *occlient.Client, kClient *kclient.Client, parameters PushParam
 
 	return nil
 }
+
+// AddEndpointInDevfile writes the provided endpoint information into devfile
+func AddEndpointInDevfile(devObj parser.DevfileObj, endpoint parsercommon.Endpoint, container string) error {
+	components := devObj.Data.GetComponents()
+	for _, component := range components {
+		if component.Container != nil && component.Container.Name == container {
+			component.Container.Endpoints = append(component.Container.Endpoints, endpoint)
+			devObj.Data.UpdateComponent(component)
+			break
+		}
+	}
+	return devObj.WriteYamlDevfile()
+}
+
+// RemoveEndpointInDevfile deletes the provided endpoint information from devfile
+func RemoveEndpointInDevfile(devObj parser.DevfileObj, urlName string, container string) error {
```

Comment:
I think we don't need the containerName since to find it we are looping over the containers and their endpoints once in GetContainerEndpoints(). Instead perform the loop here and break once the first endpoint with the name is found.

Reply:
updated